tests/src/test/java/org/eclipse/hono/tests/IntegrationTestSupport.java

```diff
@@ -542,6 +545,18 @@ private static ClientConfigProperties getClientConfigProperties(
         }
     }
 
+    private MessagingType getConfiguredMessagingType() {
+        if (getConfiguredApplicationClientType().equals(KafkaApplicationClient.class)) {
```

Comment:
IMHO we should completely replace the *getConfiguredApplicationClientType* method with this method ...

Reply:
Yes, that was already on my list as well. I refactored this in a separate PR (https://github.com/eclipse/hono/pull/2544) on which I'll rebase this one as soon as it is merged.